src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java

```diff
@@ -517,7 +518,16 @@ private String setupGlobalSettingFile() throws IOException, InterruptedException
      * @throws AbortException in case of error
      */
     private void settingsFromConfig(String settingsConfigId, FilePath settingsFile) throws AbortException {
-        Config c = Config.getByIdOrNull(settingsConfigId);
+
+        Config c = null;
+        Executor executor = Executor.currentExecutor();
+        if (executor != null) {
+            Queue.Executable currentExecutable = executor.getCurrentExecutable();
+            if (currentExecutable != null) {
+                c = ConfigFiles.getByIdOrNull((Run<?, ?>) currentExecutable, settingsConfigId);

```

Comment:
@jglick thanks, I proposed a fix on a PR on this PR: https://github.com/imod/pipeline-maven-plugin/pull/1